src/main/java/org/broadinstitute/hellbender/utils/haplotype/SAMFileDestination.java

```diff
@@ -0,0 +1,50 @@
+package org.broadinstitute.hellbender.utils.haplotype;
+
+import htsjdk.samtools.*;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+// TODO - should this use SAMFileGATKReadWriter ?
+public class SAMFileDestination extends HaplotypeBAMDestination {
+    protected SAMFileHeader samOutHeader;
+    protected SAMFileWriter samWriter;
+
+    public SAMFileDestination(File outFile, SAMFileHeader sourceHeader, String readGroupID, String haplotypeSampleTag) {
+        super(sourceHeader, readGroupID);
+
+        samOutHeader = new SAMFileHeader();
+        samOutHeader.setSequenceDictionary(samOutHeader.getSequenceDictionary());
+
+        // TODO - how should the sort order be set ?
+        //samOutHeader.setSortOrder(SAMFileHeader.SortOrder.coordinate);
+
+        final List<SAMReadGroupRecord> readGroups = new ArrayList<SAMReadGroupRecord>();
+        if (sourceHeader != null) {
+            // include the original read groups
+            readGroups.addAll(sourceHeader.getReadGroups());
+        }
+        // plus an artificial read group for the haplotypes
+        final SAMReadGroupRecord rg = new SAMReadGroupRecord(readGroupID);
+        rg.setSample(readGroupID);
```

Comment:
this is a sample name, not the read group name. This line seems wrong - it should be `rg.setSample(haplotypeSampleTag)`